Sources/Parsers/StoryboardParser.swift

```diff
@@ -28,90 +28,70 @@ public final class StoryboardParser {
-
-    @objc func parser(_ parser: XMLParser, didStartElement elementName: String,
-                      namespaceURI: String?, qualifiedName qName: String?,
-                      attributes attributeDict: [String: String]) {
-
-      switch elementName {
-      case "document":
-        initialViewControllerObjectID = attributeDict["initialViewController"]
-      case "scene":
-        inScene = true
-      case "objects" where inScene:
-        readyForFirstObject = true
-      case let tag where (readyForFirstObject && tag != "viewControllerPlaceholder"):
-        let customClass = attributeDict["customClass"]
-        let customModule = attributeDict["customModule"]
-        if let objectID = attributeDict["id"], objectID == initialViewControllerObjectID {
-          initialScene = InitialScene(objectID: objectID,
-                                      tag: tag,
-                                      customClass: customClass,
-                                      customModule: customModule)
-        }
-        if let storyboardID = attributeDict["storyboardIdentifier"] {
-          scenes.insert(Scene(storyboardID: storyboardID,
-                              tag: tag,
-                              customClass: customClass,
-                              customModule: customModule))
-        }
-        readyForFirstObject = false
-      case "connections":
-        readyForConnections = true
-      case "segue" where readyForConnections:
-        if let segueID = attributeDict["identifier"] {
-          let customClass = attributeDict["customClass"]
-          let customModule = attributeDict["customModule"]
-          segues.insert(Segue(identifier: segueID, customClass: customClass, customModule: customModule))
-        }
-      default:
-        break
-      }
-    }
 
-    @objc func parser(_ parser: XMLParser, didEndElement elementName: String,
-                      namespaceURI: String?, qualifiedName qName: String?) {
-      switch elementName {
-      case "scene":
-        inScene = false
-      case "objects" where inScene:
-        readyForFirstObject = false
-      case "connections":
-        readyForConnections = false
-      default:
-        break
+    for scene in document.xpath(XMLScene.path) {
+      guard scene.tag != XMLScene.placeholderTag else { continue }
+
+      let customClass = scene[XMLScene.customClassAttribute]
+      let customModule = scene[XMLScene.customModuleAttribute]
+
+      if scene[XMLScene.idAttribute] == initialSceneID {
+        initialScene = InitialScene(tag: scene.tag ?? "",
+                                    customClass: customClass,
+                                    customModule: customModule)
+      }
+      if let id = scene[XMLScene.storyboardIdentifierAttribute] {
+        scenes.insert(Scene(storyboardID: id,
+                            tag: scene.tag ?? "",
+                            customClass: customClass,
+                            customModule: customModule))
```

Comment:
I know we didn't consider that case in the old code either, but what if (for some reason or maybe a malformed Storyboard) we have a scene which have no `storyboardIdentifier` attribute, is it normal that we skip it? It shouldn't happen in normal cases I think, but given how Xcode generate some strange stuff with storyboards sometimes (see the issue with customModule in the other PR), maybe consider it? I don't know what would be best in that case though (ignoring it or generating a constant with an empty storyboardID?) or if that really matters or if I'm just nitpicking as often 😉